### PR TITLE
change build to check

### PR DESCRIPTION
**Pull request by Ethan-000**
# Related issue(s)

Resolves https://github.com/noir-lang/noir/issues/657

# Description

## Summary of changes

change the `nargo build` cli command to `nargo check`

## Test additions / changes

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

# Additional context

### CODE REVIEW

File: crates/nargo/src/cli/compile_cmd.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ pub(crate) fn run(args: ArgMatches) -> Result<(), CliError> {
 
     let curr_dir = std::env::current_dir().unwrap();
     let mut circuit_path = PathBuf::new();
-    circuit_path.push(BUILD_DIR);
+    circuit_path.push(CHECK_DIR);
```
Comment (jfecher) [inline crates/nargo/src/cli/compile_cmd.rs:22]: This is the path to change the output of the `compile` command, I think we should rename `CHECK_DIR` back to `BUILD_DIR` or to `COMPILE_CMD_DIR` to be more clear. The path should also be changed back to `build` or `target` for this.

**Review by jfecher — CHANGES_REQUESTED** [review]
Just one small change to revert

File: crates/nargo/src/cli/compile_cmd.rs (hunk 1 above)
Comment (Ethan-000) [reply crates/nargo/src/cli/compile_cmd.rs:22]: makes sense

File: crates/nargo/src/cli/mod.rs [hunk 2]
```diff
@@ -29,7 +29,7 @@ const VERIFIER_INPUT_FILE: &str = "Verifier";
 const SRC_DIR: &str = "src";
 const PKG_FILE: &str = "Nargo.toml";
 const PROOF_EXT: &str = "proof";
-const BUILD_DIR: &str = "build";
+const COMPILE_CMD_DIR: &str = "check";
```
[the remark this answers is not in the document]
Comment (Ethan-000) [reply crates/nargo/src/cli/mod.rs]: :ok_hand:

File: crates/nargo/src/cli/mod.rs [hunk 3]
```diff
@@ -29,7 +29,7 @@ const VERIFIER_INPUT_FILE: &str = "Verifier";
 const SRC_DIR: &str = "src";
 const PKG_FILE: &str = "Nargo.toml";
 const PROOF_EXT: &str = "proof";
-const BUILD_DIR: &str = "build";
+const TARGET_DIR: &str = "check";
```
Comment (jfecher) [inline crates/nargo/src/cli/mod.rs:32]: I think you misunderstood my suggestion, it was to change the actual directory here from `"check"` to `"target"` or `"build"`, not to change the variable name again

Comment (Ethan-000) [reply crates/nargo/src/cli/mod.rs]: ohhh, got it.

**Comment by phated** [comment]
Will we need to update any commands in the book when this change lands? cc @Globallager

**Review by jfecher — APPROVED** [review]
Looks good, thank you!

**Comment by Chisaaato** [comment]
Thanks for the review sir 🙂

**Comment by Ethan-000** [comment]
Thanks for the review sir 🙂

**Comment by Ethan-000** [comment]
@jfecher hmm should the driver function name to be changed to `check` as well or leave it as it `build` https://github.com/noir-lang/noir/blob/8c1e101a7a6e50d0a70bea5ba3ab15e53e006a78/crates/nargo/src/cli/check_cmd.rs#L28

https://github.com/noir-lang/noir/blob/348ca5dd3f1cca39ca4c89ac7520ad61bbebb940/crates/noirc_driver/src/lib.rs#L132

**Comment by jfecher** [comment]
It can be updated but there is no rush

**Pull request merged**